### PR TITLE
Feature: Implemented Accounts Feed API

### PR DESCRIPTION
**Pull request by V-R-Dighe**
- [API - Get Accounts Feed](https://github.com/TrueSparrowSystems/AI-SalesSparrow-API/issues/45)

### Summary by CodeRabbit

**Release Notes:**

- New Feature: Added support for different views in the `getAccounts` method, allowing users to switch between basic and feed views.
- New Feature: Introduced pagination in the `getAccounts` method to improve performance and user experience.
- New Feature: Enhanced the Salesforce Account DTO to include additional fields and associated contacts, providing more comprehensive account data.
- New Feature: Created a new service class `GetAccountsFeedService` to handle the retrieval of account feeds from Salesforce.
- Refactor: Updated the `GetAccountListService` to return a more detailed response including associated contacts.
- Test: Added new unit tests for encoding/decoding and JSON parsing functions in `UtilTest`.
- Test: Introduced a new functional test `GetAccountsFeedTest` for testing the account feed retrieval functionality.
- Chore: Replaced `System.out.println()` statements with logger statements in test classes for better logging practices.

### CODE REVIEW

File: src/main/java/com/salessparrow/api/lib/salesforce/helper/SalesforceQueryBuilder.java [hunk 1]
```diff
@@ -28,10 +28,24 @@ public String getAccountsQuery(String searchTerm) {
     } else {
       query = "SELECT Id, Name FROM Account WHERE Name LIKE '%"+searchTerm+"%' ORDER BY LastModifiedDate DESC LIMIT 20";
     }
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/lib/salesforce/helper/SalesforceQueryBuilder.java:30]: This functionality has been implemented.

File: src/main/java/com/salessparrow/api/services/accounts/GetAccountListService.java [hunk 2]
```diff
@@ -25,19 +27,29 @@
...
+		GetAccountsFormatterDto getAccountsFormatterDto = getAccountsFactory.getAccounts(currentUser,
+				formattedSearchString, AccountConstants.BASIC_VIEW_KIND, 0);
+
+		GetAccountListResponseDto getAccountListResponseDto = new GetAccountListResponseDto();
+		if (getAccountsFormatterDto.getAccountIds() != null) {
+			getAccountListResponseDto.setAccountIds(getAccountsFormatterDto.getAccountIds());
+		}
+		if (getAccountsFormatterDto.getAccountMapById() != null) {
+			getAccountListResponseDto.setAccountMapById(getAccountsFormatterDto.getAccountMapById());
+		}
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/services/accounts/GetAccountListService.java:46]: Invalid

File: src/main/java/com/salessparrow/api/lib/crmActions/getAccounts/GetSalesforceAccounts.java [hunk 3]
```diff
@@ -86,17 +112,40 @@
...
+			ContactEntity contactEntity = contact.getContactEntity();
+			contactMapById.put(contactEntity.getId(), contactEntity);
+			contactIds.add(contactEntity.getId());
+		}
+
+		AccountContactAssociationsEntity accountContactAssociationsEntity = new AccountContactAssociationsEntity();
+		accountContactAssociationsEntity.setContactIds(contactIds);
+		accountContactAssociationsMapById.put(salesforceAccount.getAccountEntity().getId(),
+				accountContactAssociationsEntity);
+	}
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/lib/crmActions/getAccounts/GetSalesforceAccounts.java:149]: Added todo in the code

File: src/main/java/com/salessparrow/api/services/accounts/GetAccountsFeedService.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+		User currentUser = (User) request.getAttribute("current_user");
+		pageNumber = 1;
+		String searchTerm = null;
+
+		int offset = calculateOffset(getAccountsFeedDto.getPagination_identifier());
+
+		return prepareResponse(
+				getAccountsFactory.getAccounts(currentUser, searchTerm, AccountConstants.FEED_VIEW_KIND, offset));
+	}
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/services/accounts/GetAccountsFeedService.java:54]: Added todo in the code

File: src/main/java/com/salessparrow/api/services/accounts/GetAccountsFeedService.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+			ObjectMapper mapper = new ObjectMapper();
+			String paginationJson = mapper.writeValueAsString(paginationIdentifier);
+			NextPagePayloadEntity nextPagePayload = new NextPagePayloadEntity();
+			nextPagePayload.setPaginationIdentifier(Util.base64Encode(paginationJson));
+			response.setNextPagePayload(nextPagePayload);
+		}
+		catch (Exception e) {
+			throw new CustomException(new ErrorObject("s_a_gafs_gaf_pr_1", "something_went_wrong", e.getMessage()));
+		}
+	}
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/services/accounts/GetAccountsFeedService.java:121]: Invalid

File: src/main/java/com/salessparrow/api/lib/salesforce/helper/SalesforceQueryBuilder.java [hunk 6]
```diff
@@ -32,6 +32,19 @@ public String getAccountsQuery(String searchTerm) {
...
+	 * Get the accounts feed for a given limit and offset
+	 * @param limit int
+	 * @param offset int
+	 * @return String
+	 */
+	public String getAccountFeedQuery(int limit, int offset) {
+
+		return Util.urlEncoder(String.format(
+				"SELECT Id, Name, Website, (SELECT Id, Name, Title, Email, Phone FROM Contacts) FROM Account ORDER BY LastModifiedDate ASC LIMIT %d OFFSET %d",
+				limit, offset));
```
[the remark this answers is not in the document]
Comment (V-R-Dighe) [reply src/main/java/com/salessparrow/api/lib/salesforce/helper/SalesforceQueryBuilder.java:45]: Invalid

**Review by AMAN-BARBARIA — APPROVED** [review]
LGTM

**Pull request merged**